src/FlashDriver.cpp

```diff
@@ -1,103 +1,137 @@
 #include "FlashDriver.h"
 
- // FLASH_CS is the Chip Select pin
- // 100000 is the clock speed (100kHz)
- // SPI_BITORDER_MSBFIRST sets the bit order
- // SPI_MODE1 sets the SPI mode
-FlashDriver::FlashDriver() : flashSpi(FLASH_CS, 100000, SPI_BITORDER_MSBFIRST, SPI_MODE1) {}
+
+SPIClass flashSpi(PB5, PB4, PB3, PA4);
+FlashDriver::FlashDriver() {}
 
 FlashDriver::~FlashDriver() {}
 
 FlashStatus FlashDriver::initFlash() {
 
-    if (!flashSpi.begin()) {
-        return FLASH_FAILURE;
-    }
+    uint8_t rx_buf[3] = {0}; 
+    uint8_t tx_buf = FLASH_JEDEC_ID;   
 
-    digitalWrite(FLASH_CS, LOW);
-    flashSpi.beginTransaction();
-    flashSpi.transfer(FLASH_ID_ADDR);  
+    flashSpi.begin();
 
-    // Read Manufacturer ID and Device ID
-    uint8_t manufacturerID = flashSpi.transfer(0x00);  // Read Manufacturer ID
-    uint8_t memoryType = flashSpi.transfer(0x00);      // Read Memory Type (should be 0x40 for W25Q128JV)
-    uint8_t capacity = flashSpi.transfer(0x00);        // Read Memory Capacity (should be 0x18 for W25Q128JV)
+    digitalWrite(PA4, LOW);
 
+    flashSpi.transfer(tx_buf, 1);
+    rx_buf[0] = flashSpi.transfer(0x00);
```

Comment:
Just curious what this does. 

Might be good to add more comments throughout explaining why certain things are done, so it's easier to comeback and fix things later.